### PR TITLE
Modifying resources specification to describe capabilities

### PR DESCRIPTION
**Pull request by toumorokoshi**
A focus of the specification on the capabilities that should be
afforded rather than strict requirements around implementation
enable better flexibility to ensure libraries are compatible with
the idioms of their corresponding languages.

This incorporates feedback discussed in #191.

Updates #165.

### CODE REVIEW

File: specification/api-resources.md [hunk 1]
```diff
@@ -23,54 +23,61 @@ TODO: notes how Resources API is extended when using `SDK`. https://github.com/o
...
+sources, such as environment variables, or metadata extracted from the host or 
+container.
 
-### Merge
+Already set labels MUST NOT be overwritten unless they are the empty string. 
 
-Creates a new `Resource` that is a combination of labels of two `Resource`s. For
-example, from two `Resource`s - one representing the host and one representing a
-container, resulting `Resource` will describe both.
+Label key namespacing SHOULD be used to prevent collisions across different
```
Comment (toumorokoshi) [inline specification/api-resources.md]: I wasn't clear on what his line was trying to say. Is this recommending that the labels themselves should be namespaced?

That feels like a recommendation that belongs in a section that describes a Resource object.

Comment (reyang) [reply specification/api-resources.md]: I think `Merge` API should describe the behavior when collision happened.
`Resource` concept could describe what's the recommended way of using namespaces, and provide a link to the "predefined schema".

Comment (toumorokoshi) [reply specification/api-resources.md]: Thanks! do you have an example or a link of what a predefined schema is? I'm having trouble finding that in the specification.

Comment (reyang) [reply specification/api-resources.md]: Currently there is no central place, and everything was taken from existing OpenCensus.
https://github.com/open-telemetry/opentelemetry-specification/blob/master/work_in_progress/opencensus/StandardResources.md

Comment (toumorokoshi) [reply specification/api-resources.md]: got it, thanks! As that gets merged up I'm happy to send a pr if I send see it addressed in a separate one.

**Pull request merged**